compiler/qsc_eval/src/intrinsic.rs

```diff
@@ -216,7 +253,16 @@ fn two_qubit_gate(
     if x == y {
         Err(Error::QubitUniqueness(arg_span))
     } else {
-        gate(x.unwrap_qubit().0, y.unwrap_qubit().0);
+        gate(
+            x.unwrap_qubit()
```

Comment:
You asked about code golf before -- I think some abstraction here would definitely help. Not just for code golf, but for clarity of what is going on. A wrapper struct would be easiest in my opinion. It would also provide a centralized place to dump some docstrings and overall notes on the usage of `Weak` to model qubit liveness here

Reply:
I had been thinking about our conversation on this point and added the `QubitRef` struct to wrap the notion of a weak reference to a qubit. Hopefully that, along with the docstrings it brings, helps resolve the concern here. It doesn't make the call sites much shorter, but it does make intent clearer and lets us find places that panic more easily since they are a different function call.

Reply:
Great, I am more concerned about clarity than concision here, because I know that if I was assigned a bug in these files I'd need a lot of help to know what is going on